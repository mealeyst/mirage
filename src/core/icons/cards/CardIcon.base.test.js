import React from 'react'
import { css } from 'styled-components'
import 'jest-styled-components'

import BaseCardIcon from './CardIcon.base'

const { mountWithTheme } = global

describe('(Component) BaseCardIcon', () => {
  const createBaseCardIcon = (props) => {
    return mountWithTheme(<BaseCardIcon {...props} />)
  }

  test('matching the snapshot', () => {
    expect(createBaseCardIcon())
    .toMatchSnapshot()
  })

  test('setting the className', () => {
    const className = 'example-class'
    expect(
      createBaseCardIcon({className: className}).prop('className')
    ).toEqual(className)
  })

  test('setting the children', () => {
    const children = <line x1='24' y1='0.5' x2='24' y2='3.5' />
    expect(
      createBaseCardIcon({children: children}).prop('children')
    ).toEqual(children)
  })
})
