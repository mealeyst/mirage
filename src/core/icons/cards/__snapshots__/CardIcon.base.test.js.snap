// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) BaseCardIcon matching the snapshot 1`] = `
<BaseCardIcon>
  <svg
    version="1.1"
    xmlns="http://www.w3.org/2000/svg"
  />
</BaseCardIcon>
`;
